src/search/include/search/synchronous_product.h

```diff
@@ -502,4 +494,73 @@ get_next_canonical_words(
 	return res;
 }
 
+/** Compute the direct time successors which introduce an increment in the ATA configuration for
+ * each of the passed words.
+ * @param canonical_words The set of canonical words to compute time successors of
+ * @param K The maximal constant
+ * @return All direct time successors of the passed words
+ */
+template <typename Location, typename ConstraintSymbolType>
+std::set<CanonicalABWord<Location, ConstraintSymbolType>>
+get_next_time_successors(
+  const std::set<CanonicalABWord<Location, ConstraintSymbolType>> &canonical_words,
+  RegionIndex                                                      K)
+{
+	assert(!canonical_words.empty());
+	assert(std::all_of(std::begin(canonical_words), std::end(canonical_words), [&](const auto &word) {
+		return reg_a(word) == reg_a(*std::begin(canonical_words));
+	}));
+	std::set<CanonicalABWord<Location, ConstraintSymbolType>> successors;
+	std::map<CanonicalABWord<Location, ConstraintSymbolType>,
+	         CanonicalABWord<Location, ConstraintSymbolType>>
+	  successors_map;
+	for (const auto &word : canonical_words) {
+		successors_map[word] = get_time_successor(word, K);
+	}
```

Comment:
```suggestion
	}
	bool time_successor_is_rega_equivalent = std::any_of(std::begin(successors_map), std::end(successors_map), [](const auto &map_entry) {
		    return reg_a(map_entry.first) == reg_a(map_entry.second);
	    });
```

Reply:
Probably need to run some formatter on this, the GitHub interface is limited here ;)